ledger/src/blockstore.rs

```diff
@@ -1976,10 +1985,19 @@ impl Blockstore {
         signature: Signature,
         confirmed_unrooted_slots: &[Slot],
     ) -> Result<(Option<(Slot, TransactionStatusMeta)>, u64)> {
+        // transaction_status_cf doesn't employ strong read consistency with slot-based
+        // delete_range via LedgerCleanupService, because of inefficiency.
+        // so, ensure consistent result by using lowest_cleanup_slot as the lower bound
+        // for reading
+        let lowest_cleanup_slot = self.lowest_cleanup_slot.read().unwrap();
```

Comment:
I'll write a test for this code change....

Reply:
yeah this consistency is a hard one to test 😃

Reply:
Phew, I finally wrote a test: https://github.com/solana-labs/solana/pull/16697/commits/6a296ef687da4589f5bca00c025158373e95f668